services/etcd.go

```diff
@@ -374,7 +375,7 @@ func DownloadEtcdSnapshotFromS3(ctx context.Context, etcdHost *hosts.Host, prsMa
 			fmt.Sprintf("%s:/backup:z", EtcdSnapshotPath),
 			fmt.Sprintf("%s:/etc/kubernetes:z", path.Join(etcdHost.PrefixPath, "/etc/kubernetes"))},
 		NetworkMode:   container.NetworkMode("host"),
-		RestartPolicy: container.RestartPolicy{Name: "always"},
+		RestartPolicy: container.RestartPolicy{Name: "on-failure"},
```

Comment:
Just want to confirm, this change isn't related to compression snapshot logic?

Reply:
We should be careful changing this default. I'd rather avoid doing it as a part of etcd compression pr, unless it's necessary.

Reply:
We changed the behavior of waiting for a container to exit in https://github.com/rancher/rke/pull/1301/, which means that this will keep waiting for it to exit but it won't as it will restart. Also being a one-off container and having retry logic in the code in the container, there is no need. If we don't want this change, I can run other test to validate if it will work.